module/tools/SystemConfiguration/src/SystemConfiguration.cpp

```diff
@@ -0,0 +1,363 @@
+#include "SystemConfiguration.h"
+
+#include <filesystem>
+#include <fmt/format.h>
+#include <fstream>
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <unistd.h>
+
+#include "glyphs.h"
+
+#include "extension/Configuration.hpp"
+
+namespace module::tools {
+
+using extension::Configuration;
+namespace fs = std::filesystem;
+
+bool files_equal(const fs::path& a, const fs::path& b) {
+    // If both files exist and are the same size
+    if (fs::is_regular_file(a) && fs::is_regular_file(b) && fs::file_size(a) == fs::file_size(b)) {
+        std::ifstream a_f(a);
+        std::ifstream b_f(b);
+        return std::equal(std::istreambuf_iterator<char>(a_f),
+                          std::istreambuf_iterator<char>(),
+                          std::istreambuf_iterator<char>(b_f));
+    }
+    else {
+        return false;
+    }
```

Comment:
clang-tidy would suggest this
```suggestion
    }
    return false;
```